whylog/tests/tests_config/test_get_search_range.py

```diff
@@ -46,33 +96,260 @@ def setUpClass(cls):
+        calculated_ranges = self.calculate_range(rule)
 
-        cls.rule = Rule([cause1, cause2], effect, constraints, Rule.LINKAGE_AND)
-        cls.config._rules['apache'].append(cls.rule)
+        expected_ranges = {
+            'database': {
+                'date': {
+                    'left_bound': self.one_hundred_second_earlier,
+                    'right_bound': self.ten_second_earlier
+                }
+            },
+            'apache': {
+                'date': {
+                    'left_bound': self.ten_second_earlier,
+                    'right_bound': self.effect_time
+                }
+            }
+        }
 
-    def test_get_search_range(self):
-        line_source = LineSource('localhost', 'node_1.log')
-        effect_time = datetime(2016, 4, 12, 23, 54, 43)
-        effect_line = '2016-04-12 23:54:43 effect internal server error Host: apache_host'
-        effect_clues = {'effect': Clue((effect_time,), effect_line, 40, line_source)}
+        raise SkipTest('Not implemented yet')
+        assert calculated_ranges == expected_ranges
+
+    def test_search_range_lack_of_left_bound(self):
+        constraints1 = [
+            {
+                'clues_groups': [[1, 1], [0, 1]],
+                'name': 'time',
+                'params': {'min_delta': 10}
+            },
+        ]
+        rule = Rule([self.cause1, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range(rule)
+
+        expected_ranges = {
+            'database': {
+                'date': {
+                    'left_bound': self.earliest_date,
+                    'right_bound': self.ten_second_earlier
+                }
+            },
+        }
+
+        raise SkipTest('Not implemented yet')
+        assert calculated_ranges == expected_ranges
+
+    def test_search_range_lack_of_right_bound(self):
+        constraints1 = [
+            {
+                'clues_groups': [[1, 1], [0, 1]],
+                'name': 'time',
+                'params': {'max_delta': 10}
+            },
+        ]
+        rule = Rule([self.cause1, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range(rule)
+
+        expected_ranges = {
+            'database': {
+                'date': {
+                    'left_bound': self.ten_second_earlier,
+                    'right_bound': self.effect_time
+                }
+            },
+        }
+
+        raise SkipTest('Not implemented yet')
+        assert calculated_ranges == expected_ranges
 
-        calculated_ranges = self.config._get_search_ranges([self.rule], effect_clues)
+    def test_search_range_delayed_logs(self):
+        constraints1 = [
+            {
+                'clues_groups': [[1, 1], [0, 1]],
+                'name': 'time',
+                'params': {
+                    'min_delta': -10,
```

Comment:
Warning: `searcher` implementation is not prepared yet for such case, there is a ticket for that, #131. Please don't be confused when you try to un-skip this test and it will fail.